### DISCUSSION TITLE
Nested GroupBy.groupBy

### DISCUSSION THREAD

**Discussion by macario1983** [querydsl/querydsl#3022] [Q&A]
I have 3 entities

```
public class Student {
  private Long id;
  private String name;
  private String gender;
  private Integer age;
  private List<Subject> subjects;
}

public class Subject {
  private Long id;
  private String description; 
  private List<Grade> grades;
}

public class Grade {
  private Long id;
  private BigDecimal value;
}
```

I want to generate a DTO like this:

```
public class StudentTransient {
  private Long studentId;
  private String name;
  private List<SubjectTansient> subjects;
}

public class SubjectTansient {
  private Long subjectId;
  private List<BigDecimal> grades;
}
```

So in Grade class, I just wanne get the grades and group then by subject, as I want to group subjects by student.

How can I do this?

Ps: Post a full example, because I dindt found a example like this, I used `transform` and `GroupBy.groupBy` but no achived my goal.

**Comment by leejiliang** [comment]
i have the same question, have you resolve it?

**Reply by macario1983** [reply]
> i have the same question, have you resolve it?

No, I also created question in StackOverflow and ofered bounty, but no ones have answered.
I am also waiting for them.
